components/CauseCard.tsx

```diff
@@ -1,30 +1,53 @@
 import { useTranslation } from "hooks/useTranslation";
+import { useState } from "react";
+import { updateDoc, DocumentReference } from "firebase/firestore";
 
 export const CauseCard = ({
   cause,
   type,
+  reactionsAmount,
+  documentRef,
 }: {
   cause: Cause;
   type: "help" | "seek";
+  reactionsAmount: number;
+  documentRef: DocumentReference;
 }) => {
+  const [buttonDisabled, setButtonDisabled] = useState(false);
+
   const t = useTranslation();
 
   const { descriptionId, nameId, link } = cause;
 
+  const handleClick = () => {
+    setButtonDisabled(true);
+    updateDoc(documentRef, { [cause.nameId]: reactionsAmount + 1 || 1 });
```

Comment:
Could be just nameId

```suggestion
    updateDoc(documentRef, { [nameId]: reactionsAmount + 1 || 1 });
```